<a href="https://colab.research.google.com/github/sumitvairagar/tensor-flow/blob/master/HelloWorld.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf

TensorFlow 2.x selected.


Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

---

In [0]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

Build the `tf.keras.Sequential` model by stacking layers. Choose an optimizer and loss function for training:

In [0]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train the model

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

W0816 19:32:40.320273 139693228054400 deprecation.py:323] From /tensorflow-2.0.0b1/python3.6/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 6s 94us/sample - loss: 0.2929 - accuracy: 0.9137
Epoch 2/5
60000/60000 [==============================] - 6s 92us/sample - loss: 0.1434 - accuracy: 0.9571
Epoch 3/5
60000/60000 [==============================] - 5s 90us/sample - loss: 0.1081 - accuracy: 0.9673
Epoch 4/5
60000/60000 [==============================] - 6s 92us/sample - loss: 0.0888 - accuracy: 0.9725
Epoch 5/5
10000/10000 [==============================] - 0s 45us/sample - loss: 0.0719 - accuracy: 0.9792


[0.07185291091997642, 0.9792]

The image classifier is now trained to ~98% accuracy on this dataset. To learn more, read the [TensorFlow tutorials](https://www.tensorflow.org/beta/tutorials/).